intro_to_rl.ipynb

```diff
@@ -0,0 +1,3225 @@
+{
```

Comment:
<p>This section has become extremely complicated, for something that should take only a few lines of code. The original motivation for this section was to have a method that even fairly weak students could succeed in implementing. My feeling (given previous years students) is that this complexity is beyond most students. Some simplifications:</p><ul><li>Just use NumPy to generate new random weights. Perfectly valid and much simpler than jitting some jax weight generator.</li><li>The only jitted jax is the policy/action MLP:</li></ul><pre class="ql-syntax" spellcheck="false">@jax.jit
def policy_mlp(observation, weights):
  # logits: list of length 2
  logits = jnp.dot(observation, weights)
  return jnp.argmax(logits)
</pre><p> </p><p>All the rest is just plain Python/NumPy. The very simplest is a loop that prints the average episode return every iteration, and breaks the loop if it solves cartpole (and you have immediate access to the numpy weights). </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/6/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Reply:
I have simplified this section now.